vector/src/main/java/im/vector/app/core/time/Clock.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.time
+
+import javax.inject.Inject
+
+interface Clock {
+    fun epochMillis(): Long
+}
+
+class DefaultClock @Inject constructor() : Clock {
+    override fun epochMillis(): Long {
+        return System.currentTimeMillis()
```

Comment:
A bit out of scope of this PR, but we will probably want to replace all `System.currentTimeMillis()` usages by this new Clock?

Reply:
:100: that would be the dream! I'll create an issue

Reply:
https://github.com/vector-im/element-android/issues/4562